File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/BulletFiredEvent.java
package dev.robocode.tankroyale.botapi.events;

import dev.robocode.tankroyale.botapi.BulletState;

/**
 * Event occurring when a bullet has been fired from a bot.
 */
@SuppressWarnings("unused")
public final class BulletFiredEvent extends BotEvent {

    // Bullet that was fired.
    private final BulletState bullet;

    /**
     * Initializes a new instance of the BulletFiredEvent class.
     *
     * @param turnNumber is the turn number when the bullet was fired.
     * @param bullet     is the bullet that was fired.
     */
    public BulletFiredEvent(int turnNumber, BulletState bullet) {
        super(turnNumber);
        this.bullet = bullet;
    }

    /**
     * Returns the bullet that was fired.
     *
     * @return The bullet that was fired.
     */
    public BulletState getBullet() {
        return bullet;
    }
}

File: bot-api/typescript/src/events/BulletFiredEvent.ts
import { BotEvent } from "./BotEvent.js";
import { BulletState } from "../BulletState.js";

/** Event occurring when a bullet has been fired. */
export class BulletFiredEvent extends BotEvent {
  readonly bullet: BulletState;

  constructor(turnNumber: number, bullet: BulletState) {
    super(turnNumber);
    this.bullet = bullet;
  }
}
